hail/src/main/scala/is/hail/backend/service/ServiceBackend.scala

```diff
@@ -545,10 +494,43 @@ private class HailSocketAPIOutputStream(
   }
 }
 
+case class CloudfuseConfig(bucket: String, mount_path: String, read_only: Boolean)
+
+case class SequenceConfig(fasta: String, index: String)
+
+case class ServiceBackendRPCPayload(
+  tmp_dir: String,
+  remote_tmpdir: String,
+  billing_project: String,
+  worker_cores: String,
+  worker_memory: String,
+  storage: String,
+  cloudfuse_configs: Array[CloudfuseConfig],
+  regions: Array[String],
+  flags: Map[String, String],
+  custom_references: Array[String],
+  liftovers: Map[String, Map[String, String]],
+  sequences: Map[String, SequenceConfig],
+)
+
+case class ServiceBackendExecutePayload(
+  functions: Array[SerializedIRFunction],
+  idempotency_token: String,
+  payload: ExecutePayload,
+)
+
+case class SerializedIRFunction(
+  name: String,
+  type_parameters: Array[String],
+  value_parameter_names: Array[String],
+  value_parameter_types: Array[String],
+  return_type: String,
+  rendered_body: String,
+)
+
 class ServiceBackendSocketAPI2(
```

Comment:
We should give this a better name now. It's not a socket api at all, it's just an API I guess?

Reply:
Renamed to `ServiceBackendAPI`. I could be convinced to not have this class at all and just put it on the `object ServiceBackend`